### PR TITLE
Pnomolos/default values

### PR DESCRIPTION
**Pull request by pnomolos**
Add very basic support for default values.

I borrowed some code from https://github.com/jetrockets/attrio/blob/master/lib/attrio/default_value/clonable.rb to make sure values are cloned if necessary.  I'm not sure if I correctly updated the licence information so if I haven't please let me know :)

### CODE REVIEW

File: lib/fast_attributes/builder.rb [hunk 1]
```diff
@@ -61,7 +68,7 @@ def #{attribute}=(value)
     def compile_initialize
       @methods.module_eval <<-EOS, __FILE__, __LINE__ + 1
         def initialize(attributes = {})
-          attributes.each do |name, value|
+          FastAttributes.default_attributes(self.class).merge(attributes).each do |name, value|
```
Comment (kostyantyn) [inline lib/fast_attributes/builder.rb]: When we don't have default values, we still perform a merge of 2 hashes. `{}.merge(attributes)` which will impact on a performance.

Suggest to before compiling the method, check if we have any default values and if not, iterate over attributes only.

File: lib/fast_attributes/default_attributes.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Some code borrowed from
+# https://github.com/jetrockets/attrio/blob/master/lib/attrio/default_value/clonable.rb
```
Comment (kostyantyn) [inline lib/fast_attributes/default_attributes.rb]: These lines can be removed

**Comment by kostyantyn** [comment]
Thanks for PR. The idea is really good. Let's just improve `compile_initialize` method.

Also, the old version of ruby `1.9.3` can be dropped from CI.

**Comment by pnomolos** [comment]
@kostyantyn Changes made.  I also added Ruby 2.2.5 to the travis matrix.  I guess 2.3.1 could also be added at this point.

**Comment by kostyantyn** [comment]
Looks good 👍

**Comment by pnomolos** [comment]
@kostyantyn Awesome!  Glad to help out with the project :)  Any idea when a new gem might be pushed?

**Comment by kostyantyn** [comment]
Thanks a lot for the PR. I'll push a new version in a couple of hours. Will keep you updated.

**Pull request merged**